### ISSUE TITLE
Incorrect suggestions in the common `git filter-branch --index-filter` example

### ISSUE THREAD

**Issue by borekb** [koalaman/shellcheck#1114]
The snippet below is given in the [official Git documentation](https://git-scm.com/docs/git-filter-branch#_examples) and found in various places online, I think it's correct but shellcheck gives a couple of warnings.

- Rule Id: SC2016, SC1004
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet that shows the problem:

```sh

#!/bin/bash

git filter-branch --index-filter \
  'git ls-files -s | sed "s-\t\"*-&NEWSUBDIR/-" |
   GIT_INDEX_FILE=$GIT_INDEX_FILE.new \
   git update-index --index-info &&
   mv "$GIT_INDEX_FILE.new" "$GIT_INDEX_FILE"' HEAD

```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 4:
  'git ls-files -s | sed "s-\t\"*-&NEWSUBDIR/-" |
  ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
 
Line 5:
   GIT_INDEX_FILE=$GIT_INDEX_FILE.new \
                                      ^-- SC1004: This backslash+linefeed is literal. Break outside single quotes if you just want to break the line.

$ 
```

#### Here's what I wanted or expected to see:

It should pass without any issues I believe.

**Comment by ngzhian** [comment]
I believe SC2016 is fixed by #1222
